extensions/amp-selector/1.0/selector.type.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @externs */
+
+/** @const */
+var SelectorDef = {};
+
+/**
+ * @typedef {{
+ *   as: (string|PreactDef.FunctionalComponent),
+ *   disabled: (boolean|undefined),
+ *   value: (*|undefined),
+ *   multiple: (boolean|undefined),
+ *   onChange: (?function({value: *, option: *})|undefined),
+ *   role: (string|undefined),
+ *   children: (?PreactDef.Renderable|undefined),
+ * }}
+ */
+SelectorDef.SelectorProps;
+
+/**
+ * @typedef {{
+ *   domElement: !Element,
```

Comment:
We can dedup `ShimProps` with their `Props` counterparts by adding `domElement` as a property in `Props`. I think it's OK if the non-shim components don't acknowledge it.

Reply:
I see that ShimProps and Props for selector are similar. But the option's pair seems to be much more different. I didn't really look too much into "why". Do you know?

Reply:
This is the diff. OptionShim uniquely uses: 
- `domElement`
- `isDisabled` (instead of `disabled`): There is a note here that says `isDisabled` must be used because `disabled` resolves to `undefined` for the `OptionShim` but I am not seeing any behavior to reflect this. It appears to be left from before we swapped which of `Option` and `OptionShim` was instantiated with JSX and  which was referenced with `as`. Feel free to s/`isDisabled`/`disabled` in `OptionShim` and [remove these two lines](https://github.com/ampproject/amphtml/blob/master/extensions/amp-selector/1.0/amp-selector.js#L163-L164).
- `selected`: `Option` refers to `SelectorContext` for this.

Reply:
I think it's too much delta. I'm going to keep Shim types if you don't mind. We can always clear them up later. But please follow up on the `isDisabled` stuff. And feel free to collapse them if, after that, you find them close enough.